tabbycat/results/utils.py

```diff
@@ -233,7 +234,7 @@ def send_ballot_receipt_emails_to_adjudicators(ballots, debate):
 
         scores = ''
         for team in ballot['teams']:
-            scores += _("(%(side)s) %(team)s\n") % {'side': team['side'], 'team': team['team'].short_name}
+            scores += _("(%(side)s) %(team)s\n") % {'side': team['side'], 'team': get_team_short_name(team['team'])}
```

Comment:
I like this abstraction and use of DRY, but it's more complicated than it sounds. Because `team.tournament` is called in `get_team_short_name()`, it requires the `tournament` to be fetched for each team, and then the preference (`tournament.pref('use_team_code_names')`) to be called for each team in this loop, bearing in mind that teach `team.tournament` is a different _Python_ object referring to the same _database_ object, which causes a collection of redundant database and cache lookups.

It doesn't matter so much here, because there are at most four teams, so it's like an N+1 issue where N≤4. But my current practice (until I find a better way—I'm open to suggestions) is to just call `use_team_code_names` outside the loop, and choose inside the loop which name to use, violating DRY if necessary. (In this case, if we decide to decode the names in the team point standings emails below, then DRY is a moot point.)

Reply:
I see your point.